[CI] TSLint failing

```
##[error]savegame/schemas/1001.js(49,32): error TS2339: Property 'spriteKey' does not exist on type 'StaticMapEntityComponent'.
##[error]savegame/schemas/1001.js(50,32): error TS2551: Property 'blueprintSpriteKey' does not exist on type 'StaticMapEntityComponent'. Did you mean 'getBlueprintSprite'?
##[error]savegame/schemas/1001.js(50,64): error TS2339: Property 'spriteKey' does not exist on type 'StaticMapEntityComponent'.
##[error]savegame/schemas/1001.js(56,36): error TS2551: Property 'blueprintSpriteKey' does not exist on type 'StaticMapEntityComponent'. Did you mean 'getBlueprintSprite'?
##[error]savegame/schemas/1001.js(59,36): error TS2551: Property 'blueprintSpriteKey' does not exist on type 'StaticMapEntityComponent'. Did you mean 'getBlueprintSprite'?
```

https://github.com/tobspr/shapez.io/commit/bf2eee908fedb84dbbabd359a200c446020a340e appears to break the migration from 1000 to 1001.

I don't know enough about the migrations to fix it, but I can raise a PR to at least get CI green again.

--- a/src/js/savegame/schemas/1001.js
+++ b/src/js/savegame/schemas/1001.js
@@ -1,6 +1,7 @@
 import { SavegameInterface_V1000 } from "./1000.js";
 import { createLogger } from "../../core/logging.js";
 import { T } from "../../translations.js";
+import { TypeVector, TypeNumber, TypeString, TypeNullable } from "../serialization_data_types.js";
 
 const schema = require("./1001.json");
 
@@ -43,9 +44,25 @@ export class SavegameInterface_V1001 extends SavegameInterface_V1000 {
         for (let i = 0; i < entities.length; ++i) {
             const entity = entities[i];
 
-            // FIXME - https://github.com/tobspr/shapez.io/issues/514
-            // Broken in https://github.com/tobspr/shapez.io/commit/bf2eee908fedb84dbbabd359a200c446020a340e
-            /** @type any **/
+            /**
+             * @typedef {{
+             *   origin: TypeVector,
+             *   tileSize: TypeVector,
+             *   rotation: TypeNumber,
+             *   originalRotation: TypeNumber,
+             *   spriteKey?: string,
+             *   blueprintSpriteKey: string,
+             *   silhouetteColor: string
+             * }} OldStaticMapEntity
+             */
+
+            // Here we mock the old type of the StaticMapEntity before the change to using
+            // a building ID based system (see building_codes.js) to stop the linter from
+            // complaining that the type doesn't have the properties.
+            // The ignored error is the error that the types do not overlap.  In the case
+            // of a v1000 save though, the data will match the mocked type above.
+            /** @type OldStaticMapEntity **/
+            // @ts-ignore
             const staticComp = entity.components.StaticMapEntity;
             const beltComp = entity.components.Belt;
             if (staticComp) {

